samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -94,19 +99,35 @@ void BLEConnectivityManager::ConnectionHandler(bt_conn *conn, uint8_t conn_err)
 
 void BLEConnectivityManager::DisconnectionHandler(bt_conn *conn, uint8_t reason)
 {
-	char addr[BT_ADDR_LE_STR_LEN];
+	/* Verify whether the device should be recovered */
+	BLEBridgedDevice *device = Instance().FindBLEBridgedDevice(conn);
+
+	if (device) {
+		char addr[BT_ADDR_LE_STR_LEN];
+		bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
-	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+		LOG_INF("Disconnected: %s (reason %u)", addr, reason);
 
-	LOG_INF("Disconnected: %s (reason %u)", addr, reason);
+		if (device->mConn) {
+			bt_conn_unref(device->mConn);
+			device->mConn = nullptr;
+		}
 
-	/* TODO: Implement connection re-establishment procedure. */
+		if (reason == BT_HCI_ERR_CONN_TIMEOUT) {
```

Comment:
Yes, because this reason means that we lost the connection, but we can also intentionally disconnect the device and then the recovery mechanism shouldn't start.